src/client.js

```diff
@@ -362,12 +336,16 @@ Client.prototype.inputLine = function(data) {
 	const irc = target.network.irc;
 	let connected = irc && irc.connection && irc.connection.connected;
 
-	if (Object.prototype.hasOwnProperty.call(inputs, cmd) && typeof inputs[cmd].input === "function") {
-		const plugin = inputs[cmd];
+	if (Object.prototype.hasOwnProperty.call(inputs.userInputs, cmd) && typeof inputs.userInputs[cmd].input === "function") {
+		const plugin = inputs.userInputs[cmd];
 
 		if (connected || plugin.allowDisconnected) {
 			connected = true;
-			plugin.input.apply(client, [target.network, target.chan, cmd, args]);
+			const response = plugin.input.apply(client, [target.network, target.chan, cmd, args]);
+
+			if (response && typeof response === "string") {
+				this.inputLine({target: data.target, text: response});
```

Comment:
Can you explain this to me? Plugins can return a string that would execute as a new command...?

Reply:
Yeah, so basically it's the easiest way to alias to another command if that's what we want to do.

So, if someone wants to alias `/j` to `/join`, we'd have to pass all the commands to them and they'd then have to pick out the right one and call it with the right params.

This way, we just take what they return and throw it through the parser again, re-using all the existing logic.

Reply:
I'm not a fan of putting this specific functionality here, can't the plugin call `client.inputLine` or have a better framework in place for aliasing.

Strings being interpreted as new input feels like magic to me.

Reply:
Sure, I'm happy with providing the plugin with `client.inputLine`. I'll call it something different for them, obviously. 

It's not just aliasing it's useful for. For example, if you look at the shrug command I put in the title, it just sends the message it is passed with shrug as well. So there are many other useful things this gives us, so I wouldn't want to replace this with a specific aliasing thing.

But I think that explicitly passing the `inputLine` command (with a better name) to the plugin would be a good way to solve it. I guess the issue is that we are interpreting the return value, which is very transparent magic?